### PR TITLE
react-portal: add to react-components and clean up docs

### LINKED ISSUE #25045: [Feature]: Export react-portals from react-components
### Library

React Components / v9 (@fluentui/react-components)

### Describe the feature that you would like added

react-portals is not currently exported by react-components. As a package often use when building react-component experiences, having portals included allows users to install a single package and get all of their required dependencies.

### Have you discussed this feature with our team

_No response_

### Additional context

_No response_

### Validations

- [X] Check that there isn't already an issue that request the same feature to avoid creating a duplicate.

### PR DESCRIPTION
**Pull request by micahgodbolt**
fixes #25045

### CODE REVIEW

**Comment by fabricteam** [comment]
## 📊 Bundle size report

<details>
<summary>Unchanged fixtures</summary>

| Package & Exports                                                                                                                                                             |          Size (minified/GZIP) |
| ----------------------------------------------------------------------------------------------------------------------------------------------------------------------------- | ----------------------------: |
| <samp>react-alert</samp> <br /> <abbr title='bundle-size/Alert.fixture.js'>Alert</abbr>                                                                                       |   `84.117 kB`<br />`21.07 kB` |
| <samp>react-avatar</samp> <br /> <abbr title='bundle-size/Avatar.fixture.js'>Avatar</abbr>                                                                                    |  `48.828 kB`<br />`13.842 kB` |
| <samp>react-avatar</samp> <br /> <abbr title='bundle-size/AvatarGroup.fixture.js'>AvatarGroup</abbr>                                                                          |    `14.95 kB`<br />`5.989 kB` |
| <samp>react-avatar</samp> <br /> <abbr title='bundle-size/AvatarGroupItem.fixture.js'>AvatarGroupItem</abbr>                                                                  |  `68.796 kB`<br />`19.182 kB` |
| <samp>react-combobox</samp> <br /> <abbr title='bundle-size/Combobox.fixture.js'>Combobox (including child components)</abbr>                                                 |   `74.642 kB`<br />`24.19 kB` |
| <samp>react-combobox</samp> <br /> <abbr title='bundle-size/Dropdown.fixture.js'>Dropdown (including child components)</abbr>                                                 |  `74.242 kB`<br />`24.089 kB` |
| <samp>react-components</samp> <br /> <abbr title='bundle-size/MultipleComponents.fixture.js'>react-components: Accordion, Button, FluentProvider, Image, Menu, Popover</abbr> | `189.156 kB`<br />`52.385 kB` |
| <samp>react-components</samp> <br /> <abbr title='bundle-size/ProviderAndTheme.fixture.js'>react-components: FluentProvider & webLightTheme</abbr>                            |    `33.4 kB`<br />`11.008 kB` |
| <samp>react-dialog</samp> <br /> <abbr title='bundle-size/Dialog.fixture.js'>Dialog (including children components)</abbr>                                                    |  `82.438 kB`<br />`24.444 kB` |
| <samp>react-menu</samp> <br /> <abbr title='bundle-size/Menu.fixture.js'>Menu (including children components)</abbr>                                                          |  `116.718 kB`<br />`35.78 kB` |
| <samp>react-menu</samp> <br /> <abbr title='bundle-size/Menu.Selectable.fixture.js'>Menu (including selectable components)</abbr>                                             | `119.787 kB`<br />`36.289 kB` |
| <samp>react-popover</samp> <br /> <abbr title='bundle-size/Popover.fixture.js'>Popover</abbr>                                                                                 | `102.929 kB`<br />`31.542 kB` |
| <samp>react-portal</samp> <br /> <abbr title='bundle-size/Portal.fixture.js'>Portal</abbr>                                                                                    |   `10.582 kB`<br />`3.875 kB` |
| <samp>react-portal-compat</samp> <br /> <abbr title='bundle-size/PortalCompat.fixture.js'>PortalCompatProvider</abbr>                                                         |    `5.857 kB`<br />`1.978 kB` |
| <samp>react-tooltip</samp> <br /> <abbr title='bundle-size/Tooltip.fixture.js'>Tooltip</abbr>                                                                                 |  `41.541 kB`<br />`14.646 kB` |

</details>
<sub>🤖 This report was generated against <a href='https://github.com/microsoft/fluentui/commit/79174e4d44832bed28251cde2fa1217509b2a3c2'>79174e4d44832bed28251cde2fa1217509b2a3c2</a></sub>
<div id="bundleSizeComment9423"/>

**Comment by fabricteam** [comment]
## [Perf Analysis (`@fluentui/react-components`)](https://github.com/microsoft/fluentui/wiki/Perf-Testing)
<p>No significant results to display.</p><details><summary>All results</summary><p>
  <table>
  <tr>
    <th>Scenario</th>
    <th>Render type</th>
    <th>
      <a href="https://github.com/microsoft/fluentui/wiki/Perf-Testing#why-are-results-listed-in-ticks-instead-of-time-units">Master Ticks</a>
    </th>
    <th>
      <a href="https://github.com/microsoft/fluentui/wiki/Perf-Testing#why-are-results-listed-in-ticks-instead-of-time-units">PR Ticks</a>
    </th>
    <th>Iterations</th>
    <th>Status</th>
  </tr><tr>
            <td>Avatar</td>
            <td>mount</td>
            <td><a href="https://fluentuipr.z22.web.core.windows.net/pull/25047/perf-test-react-components/Avatar-mount_base.html">1684</a></td>
            <td><a href="https://fluentuipr.z22.web.core.windows.net/pull/25047/perf-test-react-components/Avatar-mount.html">1642</a></td>
            <td>5000</td>
            <td></td>
           </tr>
<tr>
            <td>Button</td>
            <td>mount</td>
            <td><a href="https://fluentuipr.z22.web.core.windows.net/pull/25047/perf-test-react-components/Button-mount_base.html">1168</a></td>
            <td><a href="https://fluentuipr.z22.web.core.windows.net/pull/25047/perf-test-react-components/Button-mount.html">1149</a></td>
            <td>5000</td>
            <td></td>
           </tr>
<tr>
            <td>FluentProvider</td>
            <td>mount</td>
            <td><a href="https://fluentuipr.z22.web.core.windows.net/pull/25047/perf-test-react-components/FluentProvider-mount_base.html">1904</a></td>
            <td><a href="https://fluentuipr.z22.web.core.windows.net/pull/25047/perf-test-react-components/FluentProvider-mount.html">1952</a></td>
            <td>5000</td>
            <td></td>
           </tr>
<tr>
            <td>FluentProviderWithTheme</td>
            <td>mount</td>
            <td><a href="https://fluentuipr.z22.web.core.windows.net/pull/25047/perf-test-react-components/FluentProviderWithTheme-mount_base.html">715</a></td>
            <td><a href="https://fluentuipr.z22.web.core.windows.net/pull/25047/perf-test-react-components/FluentProviderWithTheme-mount.html">709</a></td>
            <td>10</td>
            <td></td>
           </tr>
<tr>
            <td>FluentProviderWithTheme</td>
            <td>virtual-rerender</td>
            <td><a href="https://fluentuipr.z22.web.core.windows.net/pull/25047/perf-test-react-components/FluentProviderWithTheme-virtual-rerender_base.html">653</a></td>
            <td><a href="https://fluentuipr.z22.web.core.windows.net/pull/25047/perf-test-react-components/FluentProviderWithTheme-virtual-rerender.html">648</a></td>
            <td>10</td>
            <td></td>
           </tr>
<tr>
            <td>FluentProviderWithTheme</td>
            <td>virtual-rerender-with-unmount</td>
            <td><a href="https://fluentuipr.z22.web.core.windows.net/pull/25047/perf-test-react-components/FluentProviderWithTheme-virtual-rerender-with-unmount_base.html">715</a></td>
            <td><a href="https://fluentuipr.z22.web.core.windows.net/pull/25047/perf-test-react-components/FluentProviderWithTheme-virtual-rerender-with-unmount.html">711</a></td>
            <td>10</td>
            <td></td>
           </tr>
<tr>
            <td>MakeStyles</td>
            <td>mount</td>
            <td><a href="https://fluentuipr.z22.web.core.windows.net/pull/25047/perf-test-react-components/MakeStyles-mount_base.html">2220</a></td>
            <td><a href="https://fluentuipr.z22.web.core.windows.net/pull/25047/perf-test-react-components/MakeStyles-mount.html">2260</a></td>
            <td>50000</td>
            <td></td>
           </tr>
<tr>
            <td>SpinButton</td>
            <td>mount</td>
            <td><a href="https://fluentuipr.z22.web.core.windows.net/pull/25047/perf-test-react-components/SpinButton-mount_base.html">3195</a></td>
            <td><a href="https://fluentuipr.z22.web.core.windows.net/pull/25047/perf-test-react-components/SpinButton-mount.html">3218</a></td>
            <td>5000</td>
            <td></td>
           </tr></table></p></details>

<div id="perfComment9425"/>

**Review by khmakoto — APPROVED** [review]

**Comment by ling1726** [comment]
Have we gotten requests to publish react-portal from the suite package ?

**Comment by micahgodbolt** [comment]
@ling1726 yes, we had a request. There is a team already using it via the react-portal package, and there is confusion as to why it's not available in the primary package.

File: packages/react-components/react-portal/src/stories/Portal/PortalDefault.stories.tsx [hunk 1]
```diff
@@ -16,33 +14,20 @@ const useStyles = makeStyles({
...
+        <Portal mountNode={rootElement}>
+          <div className={styles.portalContent}>Portal content</div>
+          <Portal mountNode={rootElement}>
+            <div className={styles.portalContent}>Portal within a Portal content</div>
+          </Portal>
         </Portal>
-      </Container>
-    </Container>
+      </div>
+    </div>
```
Comment (layershifter) [inline packages/react-components/react-portal/src/stories/Portal/PortalDefault.stories.tsx:31]: Let's avoid nesting in a default example and make it simpler:

```tsx
const useStyles = makeStyles({
  content: {
    backgroundColor: tokens.colorPaletteYellowBackground3,
    ...shorthands.border('3px', 'dashed'),
    ...shorthands.padding('5px'),
  },
});

export const Default = () => {
  const styles = useStyles();
  const [rootElement, setRootElement] = React.useState<HTMLElement | null>(null);

  return (
    <>
      <Portal mountNode={rootElement}>
        <div className={styles.content}>Portal content</div>
      </Portal>

      <div ref={setRootElement} />
    </>
  );
};
```

File: change/@fluentui-react-components-4e2aeb16-c261-4fa6-b46a-cc073bba1997.json [hunk 2]
```diff
@@ -0,0 +1,7 @@
+{
+  "type": "minor",
+  "comment": "chore: Add react-portal to react-components package",
```
Comment (layershifter) [inline change/@fluentui-react-components-4e2aeb16-c261-4fa6-b46a-cc073bba1997.json:3]:
```suggestion
  "comment": "feat: Re-export Portal component and related bits",
```

_nit_

File: change/@fluentui-react-portal-db1a41bd-bcbd-4135-a22e-9a04232064c4.json [hunk 3]
```diff
@@ -0,0 +1,7 @@
+{
+  "type": "patch",
```
Comment (layershifter) [inline change/@fluentui-react-portal-db1a41bd-bcbd-4135-a22e-9a04232064c4.json:2]:
```suggestion
  "type": "none",
```

It's not affecting published artifacts in any way, so "none" is more reasonable

**Review by layershifter — APPROVED** [review]

**Review by TristanWatanabe — APPROVED** [review]

**Review by ling1726 — APPROVED** [review]

**Pull request merged**